File: app/core/schemas/project.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from .user import UserRead


class ProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    org_id: uuid.UUID
    name: str
    description: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime
    creator: UserRead | None


class UserProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    user_id: uuid.UUID
    project_id: uuid.UUID
    created_at: datetime
    user: UserRead


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None

File: frontend-site/src/types/project.ts
export interface ProjectCreator {
  id: string
  email: string
  display_name: string
}

export interface Project {
  id: string
  org_id: string
  name: string
  description: string | null
  is_active: boolean
  created_at: string
  updated_at: string
  creator: ProjectCreator | null
}

export interface ProjectMember {
  user_id: string
  project_id: string
  created_at: string
  user: ProjectCreator
}